sdk/python/tests/compiler/compiler_tests.py

```diff
@@ -548,6 +549,24 @@ def test_tolerations(self):
 
     self._test_op_to_template_yaml(op1, file_base_name='tolerations')
 
+  def test_affinity(self):
+    """Test a pipeline with a affinity."""
+    affinity = V1Affinity(
+      node_affinity=V1NodeAffinity(
+        required_during_scheduling_ignored_during_execution=V1NodeSelector(
+          node_selector_terms=[V1NodeSelectorTerm(match_expressions=[V1NodeSelectorRequirement(
+            key='beta.kubernetes.io/instance-type', operator='In', values=['p2.xlarge'])])])))
+
+    op1 = dsl.ContainerOp(
+      name='download',
+      image='busybox',
+      command=['sh', '-c'],
+      arguments=['sleep 10; wget localhost:5678 -O /tmp/results.txt'],
+      file_outputs={'downloaded': '/tmp/results.txt'}) \
+      .add_affinity(affinity)
+
+    self._test_op_to_template_yaml(op1, file_base_name='affinity')
```

Comment:
Can you please do a feature behavior test here similar to `test_set_retry` (as opposed to just comparing the whole YAML)? https://github.com/kubeflow/pipelines/blob/1fd29cd235ee0ce50b3ca8907f094cc5c5546e99/sdk/python/tests/compiler/compiler_tests.py#L347..L358

Reply:
Added.